### ISSUE TITLE
Shadow Priest Talent Masochoism (Broken)

### ISSUE THREAD

**Issue by ghost** [gamefreedomgit/Maelstrom#3552]
[//]: # (REMBEMBER! Add links to things related to the bug using for example:)
[//]: # (http://wowhead.com/)
[//]: # (cata-twinhead.twinstar.cz)

**Description:**
Shadow Priest talent "Masochism" needs to be fixed. It doesn't proc off sd:d like it's supposed to. It procs off of Mind Blast, but will not proc from Mind Blast if you're using Mind Melt procs to cast MB. It's also not consistent, getting the replenishment  proc from Masochism is random it seems.

**How to reproduce:**

**How it should work:**
Should proc off of Shadow Word: Death self damage and receiving damage from a any source that is greater or equal to 10% of your health 

**Database links:**

**Comment by Sketish** [comment]
Please provide snaps that using SW:D is not generating mana

**Comment by warmasterhorus** [comment]
[](url)
![mb proc](https://github.com/gamefreedomgit/Maelstrom/assets/128514808/61c04cad-a662-4db2-8ac2-b87399f4cee2)
![swd no proc](https://github.com/gamefreedomgit/Maelstrom/assets/128514808/cbae03ab-5070-4d29-94a6-f281d439a3ca)
![mb no pro](https://github.com/gamefreedomgit/Maelstrom/assets/128514808/dc96ce61-a38f-456f-89a2-a71ef9fb4b09)

Here are 3 screenshots from a recording I took, but was unable to upload do to its size. it shows sw:d hitting target, damaging me and not triggering masochism, using mind blast not triggering masochism, and finally mind blast triggering masochism. getting masochism to proc is random and rare. i did another recording where I casted mind blast 4 times before I got masochism to trigger. Again it will never trigger if you use Mind Blast off of Mind Melt. You cannot get it to trigger from SW:D. It will not and does not. It will only proc off Mind Blast. I have been testing this for a long time. Test Servers and Live.
